apps/studio/src/server/modules/resource/resource.service.ts

```diff
@@ -156,42 +163,52 @@ export const updatePageById = (
 }
 
 export const updateBlobById = async (
-  db: SafeKysely,
-  props: {
+  tx: Transaction<DB>,
+  {
+    pageId,
+    content,
+    siteId,
+  }: {
     pageId: number
-    content: PrismaJson.BlobJsonContent
+    content: UnwrapTagged<PrismaJson.BlobJsonContent>
     siteId: number
   },
 ) => {
-  const { pageId: id, content } = props
-  const page = await db
+  const page = await tx
     .selectFrom("Resource")
-    .where("Resource.id", "=", String(id))
-    .where("siteId", "=", props.siteId)
+    .where("Resource.id", "=", String(pageId))
+    .where("siteId", "=", siteId)
     // NOTE: We update the draft first
     // Main should only be updated at build
     .select("draftBlobId")
-    .executeTakeFirstOrThrow()
+    .executeTakeFirst()
+
+  if (!page) {
+    throw new TRPCError({
+      code: "NOT_FOUND",
+      message: "Resource not found",
+    })
+  }
```

Comment:
no real opinion here - noticed previous call didn't throw error but this now does. not sure how muych value consistency brings here but if nextjs has good error handling layer, i think it's ok to throw in services